### PR TITLE
isViewFileSystem method should check child FS for CachingFileSystem type

### PR DESCRIPTION
**Pull request by chliang71**
Currently isViewFileSystem method only checks if the fs instance itself is a ViewFileSystem type. However when using CachingFileSystem, it is possible that the fs instance itself is CachingFileSystem (not ViewFileSystem), but its underlying child FS is still a ViewFileSystem instance. In this case it should be treated as a ViewFileSystem. This is important because ingestion staging directory needs to be relative path instead of /tmp for ViewFileSystem

Test plan - (Please fill in how you tested your changes)

Tested in our internal testing environment

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by chliang71** [comment]
@beinan mind to take a look? thanks

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java [hunk 1]
```diff
@@ -394,7 +395,22 @@ public static boolean isS3FileSystem(HdfsContext context, HdfsEnvironment hdfsEn
     public static boolean isViewFileSystem(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
     {
         try {
-            return getRawFileSystem(hdfsEnvironment.getFileSystem(context, path)) instanceof ViewFileSystem;
+            FileSystem rawFileSystem = getRawFileSystem(hdfsEnvironment.getFileSystem(context, path));
```
Comment (beinan) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:398]: is it possible to put the logic in getRawFileSystem?   something like below. 
```
    private static FileSystem getRawFileSystem(FileSystem fileSystem)
    {
        if (fileSystem instanceof HadoopExtendedFileSystem) {
            return getRawFileSystem(((HadoopExtendedFileSystem) fileSystem).getRawFileSystem());
        } else if (fileSystem instanceof CachingFilesystem) {
            return getRawFileSystem(cachingfilesystem.getDatatier())
        }
        return fileSystem;
    }
```

**Review by beinan — COMMENTED** [review]
Thank you @chliang71 !  Just one comment in the pr

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java (hunk 1 above)
Comment (beinan) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java:398]: Ok, I think this has been implemented in https://github.com/prestodb/presto/pull/17390/files

**Comment by chliang71** [comment]
Closing this PR as the similar functionality has been recently introduced by https://github.com/prestodb/presto/pull/17390/files

**Pull request closed without merging**